GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -1294,45 +1286,32 @@ void DoCommit()
                         AllowCancel = true,
                     };
                     TaskDialogCommandLinkButton btnCheckout = new(TranslatedStrings.ButtonCheckoutBranch);
-                    btnCheckout.Click += (s, e) =>
-                    {
-                        dialogResult = 0;
-                    };
                     TaskDialogCommandLinkButton btnCreate = new(TranslatedStrings.ButtonCreateBranch);
-                    btnCreate.Click += (s, e) =>
-                    {
-                        dialogResult = 1;
-                    };
                     TaskDialogCommandLinkButton btnContinue = new(TranslatedStrings.ButtonContinue);
-                    btnContinue.Click += (s, e) =>
-                    {
-                        dialogResult = 2;
-                    };
                     page.Buttons.Add(btnCheckout);
                     page.Buttons.Add(btnCreate);
                     page.Buttons.Add(btnContinue);
 
-                    TaskDialog.ShowDialog(Handle, page);
-
-                    switch (dialogResult)
+                    TaskDialogButton result = TaskDialog.ShowDialog(Handle, page);
+                    if (result == TaskDialogButton.Cancel)
```

Comment:
`if (result == TaskDialogButton.Cancel)`
to
if (result is TaskDialogButton.Cancel)

Reply:
It will not compile. `TaskDialogButton.Cancel` is not a constant value.

Reply:
Are you sure? We designed it to allow this scenario